### ISSUE TITLE
New Resource: azurerm_data_lake_store_virtual_network_rule

### ISSUE THREAD

**Issue by philbal611** [terraform-providers/terraform-provider-azurerm#9157]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Allows virtual network rules to be written on Data Lake Stores, as available through the Data Lake service.

### New or Affected Resource(s)

* azurerm_data_lake_store_virtual_network_rule

### Potential Terraform Configuration

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "northeurope"
}

resource "azurerm_virtual_network" "vnet" {
  name                = "example-vnet"
  address_space       = ["10.7.29.0/29"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "subnet" {
  name                 = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.vnet.name
  address_prefixes     = ["10.7.29.0/29"]
  service_endpoints    = ["Microsoft.AzureActiveDirectory"] # required
}

resource "azurerm_data_lake_store" "example" {
  name                = "consumptiondatalake"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
}

resource "azurerm_data_lake_store_virtual_network_rule" "example" {
  name                = "office-vnet-rule"
  account_name        = azurerm_data_lake_store.example.name
  resource_group_name = azurerm_resource_group.example.name
  subnet_id           = azurerm_subnet.subnet.id
}
```

### References

* [MS doc for enabling virtual network rule on Data Lake Store](https://docs.microsoft.com/en-us/azure/data-lake-store/data-lake-store-network-security#step-2-set-up-the-allowed-virtual-network-or-subnet-for-your-data-lake-storage-gen1-account)

**Comment by brunhil** [comment]
I'll take a look at this one!

**Comment by jbinko** [comment]
@brunhil - just created PR #10430 for this. I was working on this for couple of last days. Sorry for not informing about this earlier.

**Comment by jackofallops** [comment]
This appears to have been resolved by 10430 and released in provider version `v2.56.0` so I'm going to close this issue.
